vllm/model_executor/model_loader.py

```diff
@@ -1,16 +1,107 @@
 """Utilities for selecting and loading models."""
 import contextlib
+import time
 from typing import Type
 
 import torch
-import torch.nn as nn
 from transformers import PretrainedConfig
 
+from vllm.logger import init_logger
 from vllm.config import ModelConfig
 from vllm.model_executor.models import ModelRegistry
 from vllm.model_executor.weight_utils import (get_quant_config,
                                               initialize_dummy_weights)
 
+from torch import nn
+
+from tensorizer import TensorDeserializer, TensorSerializer, stream_io
+from tensorizer.utils import convert_bytes, get_mem_usage, no_init_or_tensor
+
+logger = init_logger(__name__)
+
+
+class TensorizerAgent:
+
+    def __init__(self, model_cls: Type[nn.Module], model_config: ModelConfig):
+        self.model_config = model_config
+        self.model_cls = model_cls
+        self.tensorizer_args = self.model_config.tensorizer_args
+        self.serialize_args = self.tensorizer_args.serializer_params
+        self.deserialize_args = self.tensorizer_args.deserializer_params
+
+        self.serialize_model = not self._verify_path_reachable()
+
+    def _verify_path_reachable(self):
+        if not self.tensorizer_args.download_dir.endswith(".tensors"):
+            raise ValueError(f"download_dir {self.tensorizer_args.download_dir} must specify a .tensors "
+                             f"file when load_format = tensorizer")
+        try:
+            stream_io.open_stream(self.tensorizer_args.download_dir, "rb")
+            return True
+        except OSError as err:
+            if "Not Found" in str(err):
+                logger.info(
+                    f"Tensors not found. Will load via HF and serialize tensors to {self.tensorizer_args.download_dir}"
+                )
+                return False
+            else:
+                raise OSError(err)
+
+    def serialize(self):
+        with torch.device("cuda"):
+            model = self.model_cls(self.model_config.hf_config)
+        self.model_config.load_format = "auto"
```

Comment:
Were you able to try `fp16` or anything that is not `auto`? What was the issue here?

Reply:
`load_format` does not specify the `dtype` of the loaded model. It specifies the way in which model weights should be loaded, whether that's using `auto`, `pt`, `safetensors`, `npcache`, `dummy`, and now `tensorizer`. Seeing as in the case of serializing model weights if no weights can be found at `download_dir`, this logic is to support loading the model through other means to then serialize it. `auto` is the default load format option, and seemed appropriate here. It attempts to load the model using `safetensors`, and failing that, through `pt`.